packages/compiler/core/parser.ts

```diff
@@ -279,10 +290,11 @@ export function parse(code: string | SourceFile, options: ParseOptions = {}): Ca
 
   function parseCadlScript(): CadlScriptNode {
     const statements = parseCadlScriptItemList();
-    return {
+    return createNode({
       kind: SyntaxKind.CadlScript,
       statements,
       file: scanner.file,
+      id: createNode({ kind: SyntaxKind.Identifier, sv: scanner.file.path, pos: 0, end: 0 }),
       namespaces: [],
       usings: [],
       locals: createSymbolTable(),
```

Comment:
So locals still remains on the node itself? Does that not feel a bit inconsistent with exports?

Reply:
It is inconsistent but locals and exports (and soon members) are not the same thing, which is part of why I like the change.

Locals are only ever accessed off of the node, in a context of doing an upwards walk during scope resolution. Exports are accessed outside the context of a node (e.g. when dotting into, or merging symbols). It is never convenient to have locals hanging off the symbol, while it is often convenient for exports to exist on the symbol.

I think this is a reflection of exports really being about the symbol, while locals are really about the node.